### PR TITLE
feat: add TileWithButton component

### PR DESCRIPTION
**Pull request by rosvik**
closes https://github.com/AtB-AS/kundevendt/issues/16297

Adds a new "TileWithButton" component, that is used for recent face contracts, and soon for bundled benefits in ticket details.

### CODE REVIEW

**Review by hanne-at-atb — COMMENTED** [review]
Sounds and looks nice, should it be added to Storybook too while you're at it? 🏬

**Review by marius-at-atb — APPROVED** [review]
Looks good!

**Comment by rosvik** [comment]
> @hanne-at-atb  Sounds and looks nice, should it be added to Storybook too while you're at it? 🏬

I started looking at it, but had some problems with supplying some of the more complex props (ThemeIcon / children ++). I can try to come up with a limited example, and include that 👍

File: src/components/tile/TileWithButton.tsx [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+        ]}
+        testID={testID + 'BuyButton'}
+      >
+        <ThemeText
+          style={styles.buttonText}
+          type={mode === 'spacious' ? 'body__primary' : 'body__tertiary'}
+        >
+          {buttonText}
+        </ThemeText>
+        <ThemeIcon size="small" svg={buttonSvg} fill={color.outline.text} />
```
[the remark this answers is not in the document]
Comment (rosvik) [reply src/components/tile/TileWithButton.tsx:71]: How did I not notice that 🙈

File: src/storybook/stories/TileWithButton.stories.tsx [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+    accessibilityLabel: 'Accessibility label',
+    buttonText: 'Button text',
+    buttonSvg: SvgArrowRight,
+    onPress: () => {},
+    ...themedStoryDefaultArgs,
+  },
+  decorators: [
+    (_: any, {args}: {args: any}) => (
+      <View style={{justifyContent: 'center', flexDirection: 'row'}}>
+        {/* FIXME: Should use <Story> here, but that doesn't work well with children */}
```
Comment (rosvik) [inline src/storybook/stories/TileWithButton.stories.tsx:46]: Open for suggestions here 😅

**Review by hanne-at-atb — APPROVED** [review]
🍾

**Pull request merged**